### PR TITLE
Added support for showExtensions and $ref for Responses

### PR DESCRIPTION
**Pull request by shelbys**
If you appreciate this contribution, please add a `hacktoberfest-accepted` Label to the PR, so it'll be tracked

### CODE REVIEW

File: src/services/__tests__/models/Response.test.ts [hunk 1]
```diff
@@ -31,5 +31,12 @@ describe('Models', () => {
       const resp = new ResponseModel(parser, 'default', true, {}, opts);
       expect(resp.type).toEqual('error');
     });
+
+    test('should be error if showExtensions is true', () => {
```
Comment (RomanHotsiy) [inline src/services/__tests__/models/Response.test.ts:35]: should be error?

**Review by RomanHotsiy — CHANGES_REQUESTED** [review]
It looks good, the only question is if the test message is corret?

File: src/services/__tests__/models/Response.test.ts (hunk 1 above)
Comment (gileswells) [inline src/services/__tests__/models/Response.test.ts:35]: Maybe this?

```suggestion
    test('ensure extensions are shown if showExtensions is true', () => {
```

**Review by RomanHotsiy — APPROVED** [review]

File: src/services/__tests__/models/Response.test.ts (hunk 1 above)
Comment (shelbys) [reply src/services/__tests__/models/Response.test.ts]: Oops!  I missed updating the first half.  ;)

**Comment by shelbys** [comment]
Thanks much, including for the tagging

**Comment by shelbys** [comment]
@RomanHotsiy - Please merge this when you get a chance to unblock #1434

**Comment by RomanHotsiy** [comment]
Hey @shelbys!

I am very sorry for the 1-year delay! 🙈
We've been busy with other projects and I missed this PR (and many more others 😢 ).

We now have more resources and time to work on Redoc so this PR will be merged today.

**Pull request merged**